[MI-2502]:Added custom type for webhook post in order to comment, edit and close issues

Nityanand13 opened the pull request.

Nityanand13 committed:
[MI-2502]:Added custom type for webhook post in order to comment, edit and close issues

Nityanand13 committed:
[MI-2502]: Added EOF

Nityanand13 committed:
[MI-2502]: Fixed lint errors

Kshitij-Katiyar commented on go.mod:
```diff
@@ -5,6 +5,7 @@ go 1.16
 require (
 	github.com/Masterminds/sprig/v3 v3.2.2
 	github.com/google/go-github/v41 v41.0.0
+	github.com/google/go-github/v48 v48.2.0
```
Have you updated the new package everywhere if it is we can remove the previous one having 2 versions is not a good practice

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -184,8 +190,8 @@ func (p *Plugin) checkConfigured(next http.Handler) http.Handler {
 	return func(w http.ResponseWriter, r *http.Request) {
-		userID := r.Header.Get("Mattermost-User-ID")
-		if userID == "" {
+		HeaderMattermostUserID := r.Header.Get("Mattermost-User-ID")
```
see this again whether it is correct or not

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	userID := r.Header.Get(HeaderMattermostUserID)
+	post, appErr := p.API.GetPost(postID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + postID, StatusCode: http.StatusInternalServerError})
```
Check this everywhere
```suggestion
		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load the post " + postID, StatusCode: http.StatusInternalServerError})
```

Kshitij-Katiyar replied:
use fmt.sprintf for concentation

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	owner := r.FormValue("owner")
+	repo := r.FormValue("repo")
+	number := r.FormValue("number")
+	postID := r.FormValue("postId")
```
constants for all

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + postID + ": not found", StatusCode: http.StatusNotFound})
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	if post == nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + postID + ": not found", StatusCode: http.StatusNotFound})
+		return
+	}
```
```suggestion
	}
	
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	owner := r.FormValue("owner")
+	repo := r.FormValue("repo")
+	number := r.FormValue("number")
+	status := r.FormValue("status")
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	repo := r.FormValue("repo")
+	number := r.FormValue("number")
+	postID := r.FormValue("postId")
+	numberInt, err := strconv.Atoi(number)
```
numberInt doesnt make sense here please change it

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	issue, _, err := githubClient.Issues.Get(c.Ctx, owner, repo, numberInt)
+	if err != nil {
+		// If the issue is not found, it's probably behind a private repo.
+		// Return an empty repose in this case.
```
```suggestion
		// Return an empty response in that case.
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+
+	issue, _, err := githubClient.Issues.Get(c.Ctx, owner, repo, numberInt)
+	if err != nil {
+		// If the issue is not found, it's probably behind a private repo.
```
```suggestion
		// If the issue is not found, it probably belongs to a private repo.
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+				"owner":  owner,
+				"repo":   repo,
+				"number": numberInt,
+			}).Debugf("Issue  not found")
```
```suggestion
			}).Debugf("Issue not found")
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+			"owner":  owner,
+			"repo":   repo,
+			"number": numberInt,
+		}).Debugf("Could not get issue")
```
```suggestion
		}).Debugf("Could not get the issue")
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+			"repo":   repo,
+			"number": numberInt,
+		}).Debugf("Could not get issue")
+		p.writeAPIError(w, &APIErrorResponse{Message: "Could not get issue", StatusCode: http.StatusInternalServerError})
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -139,6 +140,11 @@ func (p *Plugin) initializeAPI() {
 	apiRouter.HandleFunc("/createissue", p.checkAuth(p.attachUserContext(p.createIssue), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/closeorreopenissue", p.checkAuth(p.attachUserContext(p.closeOrReopenIssue), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/updateissue", p.checkAuth(p.attachUserContext(p.updateIssue), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/editissuemodal", p.checkAuth(p.attachUserContext(p.openIssueEditModal), ResponseTypePlain)).Methods(http.MethodGet)
```
whats the reason for all the below APIs are "GET" ?

Nityanand13 replied:
I didn't get that question. I mean these are "GET" because I want to fetch the data.

Kshitij-Katiyar replied:
are u actually fetching the data here in all the below APIs? check again

Nityanand13 replied:
To whichever i have added a get request, I am fetching the data and publishing a web socket event.

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	}
+
+	assignees := make([]string, len(issue.Assignees))
+	for i, v := range issue.Assignees {
```
have a better name for both i and v and check that everywhere

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,6 +1044,161 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	userID := r.Header.Get(HeaderMattermostUserID)
+	post, appErr := p.API.GetPost(postID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + postID, StatusCode: http.StatusInternalServerError})
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
 }
 
+func (p *Plugin) updateIssue(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	type IssueRequest struct {
```
always create a global struct

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		IssueNumber int      `json:"issue_number"`
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
```
see if you can make this better or more detailed that would be better

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding JSON body")
```
```suggestion
		c.Log.WithError(err).Warnf("Error decoding the JSON body")
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding JSON body")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
```
i think this error message is not correct as the input can be a json and we can still get this error

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		return
+	}
+
+	if issue.Title == "" {
```
create a separate function for validation of the body

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		var appErr *model.AppError
+		post, appErr = p.API.GetPost(issue.PostID)
+		if appErr != nil {
+			p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + issue.PostID, StatusCode: http.StatusInternalServerError})
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+			return
+		}
+		permalink = p.getPermaLink(issue.PostID)
+	}
```
```suggestion
	}
	
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		}
+		permalink = p.getPermaLink(issue.PostID)
+	}
+	ghIssue := &github.IssueRequest{
```
```suggestion
	githubIssue := &github.IssueRequest{
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		return
+	}
+
+	splittedRepo := strings.Split(issue.Repo, "/")
```
check the length of splittedRepo here before accessing indexes

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	result, resp, err := githubClient.Issues.Edit(c.Ctx, owner, repoName, issue.IssueNumber, ghIssue)
+	if err != nil {
+		if resp != nil && resp.Response.StatusCode == http.StatusGone {
+			p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Issues are disabled on this repository.", StatusCode: http.StatusMethodNotAllowed})
```
i think the status code is not correct here

Nityanand13 replied:
For this case, they have earlier written this status code only. So, I got that from there.

Kshitij-Katiyar replied:
no problem even if is correct give the reason here if not then change it

Nityanand13 replied:
I explored an found that it is correct as we encouter `StatusGone` only when access to the target resource is no longer available like if the page has been deleted. `StatusMethodNotAllowed` indicates that the server received and recognized the specified request HTTP method, but the server rejected that particular method for the requested resource.

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+			p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Issues are disabled on this repository.", StatusCode: http.StatusMethodNotAllowed})
+			return
+		}
+		c.Log.WithError(err).Warnf("Failed to update issue")
```
```suggestion
		c.Log.WithError(err).Warnf("Failed to update the issue")
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+			return
+		}
+		c.Log.WithError(err).Warnf("Failed to update issue")
+		p.writeAPIError(w,
+			&APIErrorResponse{
```
```suggestion
		p.writeAPIError(w, &APIErrorResponse{
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		c.Log.WithError(err).Warnf("Failed to update issue")
+		p.writeAPIError(w,
+			&APIErrorResponse{
+				ID: "",
```
any reason why we are sending empty ID ?

Nityanand13 replied:
I think we don't have ID field in the response, or there might be another reason. By the way I have done this because for this case, they have earlier sent empty ID only. So, I got that from there.

Kshitij-Katiyar replied:
still try to find the reason for that and why it is imp

Nityanand13 replied:
Basically with the help of this ID we check what type of error it is.
Please look at here:

https://github.com/mattermost/mattermost-plugin-gitlab/blob/57e74ce66f8f6f4f59633813663a84b2e0a22191/server/plugin.go#L263

https://github.com/mattermost/mattermost-plugin-gitlab/blob/57e74ce66f8f6f4f59633813663a84b2e0a22191/server/api.go#L465

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + issue.PostID + ": not found", StatusCode: http.StatusNotFound})
+		return
+	}
+	post.Props["assignees"] = issue.Assignees
```
constants for this

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	p.writeJSON(w, result)
+}
+
+func (p *Plugin) CreateCommentToIssue(c *UserContext, w http.ResponseWriter, comment string, owner string, repo string, post *model.Post, numberInt int) {
```
Better naming than numberInt
```suggestion
func (p *Plugin) CreateCommentToIssue(c *UserContext, w http.ResponseWriter, comment, owner, repo string, post *model.Post, numberInt int) {
```

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Improved code quality
2. Made few contants

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -27,8 +27,25 @@ const (
+	IssueNumberForProps = "issue_number"
+	IssueUrlForProps    = "issue_url"
+	RepoOwnerForProps   = "repo_owner"
+	RepoNameForProps    = "repo_name"
```
Its better to group the constants like props and params etc
```suggestion
	RepoNameForProps    = "repo_name"

```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -62,6 +79,79 @@ type Context struct {
+	ShowAttachedMessage bool   `json:"show_attached_message"`
+}
+
+// Only send down fields to client that are needed
```
```suggestion
// Only send down fields to the client that are needed
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -62,6 +79,79 @@ type Context struct {
+	PostID       string `json:"postId"`
+}
+
+type IssueRequestToCreate struct {
```
Is this better ?
```suggestion
type CreateIssueRequest struct {
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -936,7 +998,9 @@ func (p *Plugin) createIssueComment(c *UserContext, w http.ResponseWriter, r *ht
 
-	req.Comment = permalinkMessage + req.Comment
+	if req.ShowAttachedMessage {
+		req.Comment = permalinkMessage + req.Comment
```
always try to use fmt.Sprintf for string concentation. check this everywhere

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,39 +1099,196 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+	if err != nil {
+		p.writeAPIError(w, &APIErrorResponse{Message: "Invalid param 'number'.", StatusCode: http.StatusBadRequest})
+		return
+	}
```
```suggestion
	}
	
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1035,39 +1099,196 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+			"assignees":        assignees,
+			"labels":           labels,
+			"description":      description,
+			"repo_full_name":   owner + "/" + repo,
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	post.Props[DescriptionForProps] = issue.Body
+	post.Props[TitleForProps] = issue.Title
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to update post " + issue.PostID, StatusCode: http.StatusInternalServerError})
```
```suggestion
		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to update the post " + issue.PostID, StatusCode: http.StatusInternalServerError})
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	issue := &IssueRequestToUpdate{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding the JSON body")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid JSON object.", StatusCode: http.StatusBadRequest})
```
do we need to send the actual error as well to the API response?

Nityanand13 replied:
I don't this soo.. as we are sending the message.

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+
+	splittedRepo := strings.Split(issue.Repo, "/")
+	if len(splittedRepo) < 2 {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repo", StatusCode: http.StatusBadRequest})
```
```suggestion
		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repository", StatusCode: http.StatusBadRequest})
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		c.Log.WithError(err).Warnf("Failed to update the issue")
+		p.writeAPIError(w, &APIErrorResponse{
+			ID: "",
+			Message: "failed to update issue: " + getFailReason(resp.StatusCode,
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		p.API.SendEphemeralPost(c.UserID, reply)
+	}
+	if appErr != nil {
+		c.Log.WithError(appErr).Warnf("failed to create notification post")
```
```suggestion
		c.Log.WithError(appErr).Warnf("failed to create the notification post")
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	p.writeJSON(w, result)
+}
+
+func (p *Plugin) CreateCommentToIssue(c *UserContext, w http.ResponseWriter, comment, owner, repo string, post *model.Post, numberInt int) {
```
better name than numberInt

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+
+	_, appErr := p.API.CreatePost(reply)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to create notification post " + post.Id, StatusCode: http.StatusInternalServerError})
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	}
+}
+
+func (p *Plugin) CloseOrReopenIssue(c *UserContext, w http.ResponseWriter, status string, statusReason string, owner string, repo string, post *model.Post, numberInt int) {
```
if i have given a comment check that for entire code I will not write same same everywhere

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		post.Props[IssueStatus] = Close
+	}
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to update post " + post.Id, StatusCode: http.StatusInternalServerError})
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+	numberInt, err := strconv.Atoi(req.Number)
```
same

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		return
+	}
+
+	_, err = p.getUsername(post.UserId)
```
Check this everywhere
```suggestion
	if _, err = p.getUsername(post.UserId); err !=nil {
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
-
-	// get data for the issue from the request body and fill IssueRequest object
-	issue := &IssueRequest{}
+	// get data for the issue from the request body and fill IssueRequestToCreate object so that we can create the issue
```
```suggestion
	// get data for the issue from the request body and fill IssueRequestToCreate to create the issue
```

Kshitij-Katiyar commented on webapp/src/action_types/index.js:
```diff
@@ -17,9 +17,11 @@ export default {
-    OPEN_CREATE_ISSUE_MODAL_WITHOUT_POST: pluginId + '_open_create_modal_without_post',
+    CLOSE_CREATE_OR_UPDATE_ISSUE_MODAL: pluginId + '_close_create_or_update_issue_modal',
+    CLOSE_CLOSE_OR_REOPEN_ISSUE_MODAL: pluginId + '_close_close_or_reopen_issue_modal',
+    OPEN_CREATE_ISSUE_MODAL_WITH_POST: pluginId + '_open_create_issue_modal_with_post',
```
do we need a close for  ```OPEN_CREATE_ISSUE_MODAL_WITH_POST```

Nityanand13 replied:
`CLOSE_CREATE_OR_UPDATE_ISSUE_MODAL` is working as close for `OPEN_CREATE_ISSUE_MODAL_WITH_POST`

Kshitij-Katiyar commented on webapp/src/components/github_issue/github_issue.jsx:
```diff
@@ -0,0 +1,117 @@
+import {makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
```
i think this has been discussed before whenever we create a new file that should be .ts and new component should be functional component. Is there any reason why this file .js ?

Kshitij-Katiyar commented on webapp/src/components/github_issue/index.js:
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+import {connect} from 'react-redux';
```
why you haven't created a functional component here ?

Kshitij-Katiyar commented on webapp/src/components/modals/close_reopen_issue/index.js:
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
```
same

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -47,11 +55,22 @@ export default class CreateIssueModal extends PureComponent {
+                value: this.props.milestoneNumber,
+                label: this.props.milestoneTitle,
+            }});
+            this.setState({issueDescription: this.props.description}); // eslint-disable-line react/no-did-update-set-state
```
we dont use setState in componentDidUpdate lifecycle method. Please explore the reason if you don't know and try to find a other method to do this

Nityanand13 replied:
I think we can do that but with a condition otherwise there will be rendering in infinite loop.

Kshitij-Katiyar replied:
rather putting on every makes a block for eslint-disable-line

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -209,6 +249,40 @@ export default class CreateIssueModal extends PureComponent {
+                        label='Title for the GitHub Issue'
+                        type='input'
+                        required={true}
+                        disabled={false}
```
i think default value for disabled is false, if it is we can remove this

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -209,6 +249,40 @@ export default class CreateIssueModal extends PureComponent {
+                        onChange={this.handleIssueTitleChange}
+                    />
+                    {issueTitleValidationError}
+
```
```suggestion
```

Kshitij-Katiyar commented on webapp/src/websocket/index.js:
```diff
@@ -85,11 +87,29 @@ export function handleRefresh(store) {
+        if (!msg.data) {
+            return;
+        }
+        store.dispatch(openCloseOrReopenIssueModal(msg.data.channel_id, msg.data.owner, msg.data.repo, msg.data.number, msg.data.status, msg.data.postId));
```
make a variable for msg.data and use that everywhere

Kshitij-Katiyar commented on webapp/src/websocket/index.js:
```diff
@@ -85,11 +87,29 @@ export function handleRefresh(store) {
+        if (!msg.data) {
+            return;
+        }
+        store.dispatch(openAttachCommentToIssueModal(msg.data.postId, msg.data.owner, msg.data.repo, msg.data.number));
```
same

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Changed few files to .tsx
2. Improved code quality

Kshitij-Katiyar commented on webapp/src/websocket/index.js:
```diff
@@ -85,11 +87,29 @@ export function handleRefresh(store) {
             return;
         }
-        store.dispatch(openCreateIssueModalWithoutPost(msg.data.title, msg.data.channel_id));
+        store.dispatch(openCreateOrUpdateIssueModal(msg.data.title, msg.data.milestone_title, msg.data.milestone_number, msg.data.issue_number, msg.data.labels, msg.data.assignees, msg.data.description, msg.data.repo_full_name, msg.data.channel_id, msg.data.postId));
```
un flur the data first and then put the values 
```
const {data} = 
```

Kshitij-Katiyar commented on webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx:
```diff
@@ -38,6 +40,24 @@ export default class AttachIssueModal extends PureComponent {
         if (!this.state.issueValue) {
+            const issue = {
+                owner: this.props.messageData.owner,
+                repo: this.props.messageData.repo,
```
un flur props first

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1465,294 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+
+	var permalinkReplyMessage string
+	switch statusReason {
+	case "completed":
```
constant for this

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1465,294 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		c.Log.WithError(err).Warnf("Failed to update the issue")
+		p.writeAPIError(w, &APIErrorResponse{
+			ID: "",
+			Message: fmt.Sprintf("failed to update issue: %s", getFailReason(resp.StatusCode,
```
```suggestion
			Message: fmt.Sprintf("failed to update the issue: %s", getFailReason(resp.StatusCode,
```

Kshitij-Katiyar commented on server/plugin/api.go:
```diff
@@ -1258,23 +1465,294 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		UserId:    c.UserID,
+	}
+
+	_, appErr := p.API.CreatePost(reply)
```
```suggestion
	if _, appErr := p.API.CreatePost(reply); err != nil {
```

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Improved code quality
2. Made a constant

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,17 +46,32 @@ export default class CreateIssueModal extends PureComponent {
 
     // handle issue creation after form is populated
     handleCreate = async (e) => {
+        const data = this.props.messageData;
```
do this everywhere wherever possible
```suggestion
        const {channel_id,issue_number} = this.props.messageData;
```

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,17 +46,32 @@ export default class CreateIssueModal extends PureComponent {
-        } else if (this.props.channelId && (this.props.channelId !== prevProps.channelId || this.props.title !== prevProps.title)) {
-            const title = this.props.title.substring(0, MAX_TITLE_LENGTH);
-            this.setState({issueTitle: title}); // eslint-disable-line react/no-did-update-set-state
+        const data = this.props.messageData;
```
same

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Improved code readability

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -62,6 +83,79 @@ type Context struct {
 	Log    logger.Logger
 }
 
+type GitHubUserRequest struct {
```
There should be a separate package for all these structs IMO.

manojmalik20 commented on webapp/src/actions/index.js:
```diff
@@ -339,28 +393,42 @@ export function updateRhsState(rhsState) {
+export function openCreateOrUpdateIssueModal(messageData) {
+    return {
+        type: ActionTypes.OPEN_CREATE_OR_UPDATE_ISSUE_MODAL,
+        data: {
+            messageData,
+        },
```
```suggestion
        data: messageData,
```
Can't we do it like this? If yes, do this in the below actions as well.

Nityanand13 replied:
No, we can't do like this

manojmalik20 replied:
Why?

Nityanand13 replied:
Here we have 2 different action type which has different info. But in reducer these action types return same state. So its not good to directly initialise `data` with `messageData`

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -27,8 +27,29 @@ const (
+	numberQueryParam       = "number"
+	postIdQueryParam       = "postId"
+
+	IssueStatus         = "status"
```
I think we should move these to a separate `constants.go` file or a separate `constants` package.

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1035,39 +1095,197 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
 }
 
+func (p *Plugin) openAttachCommentIssueModal(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	owner := r.FormValue(ownerQueryParam)
+	repo := r.FormValue(repoQueryParam)
+	number := r.FormValue(numberQueryParam)
+	postID := r.FormValue(postIdQueryParam)
```
Why aren't we taking these values in request body?

Nityanand13 replied:
Because there are only 4 fields so I passed in the query param.

manojmalik20 replied:
What if the no. of fields increase in the future? I think we should use request body here.

Nityanand13 replied:
Updated

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1035,39 +1095,197 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+		}).Debugf("Could not get the issue")
+		p.writeAPIError(w, &APIErrorResponse{Message: "Could not get the issue", StatusCode: http.StatusInternalServerError})
+		return
+	}
```
```suggestion
	}
	
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1035,39 +1095,197 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
+
+	description := ""
+	if issue.Body != nil {
+		description = *issue.Body
```
You can do this in the previous if condition on line no. 1200.

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	}
+}
+
+func (p *Plugin) validateIssueRequestToUpdate(issue *UpdateIssueRequest, w http.ResponseWriter) bool {
```
```suggestion
func (p *Plugin) validateIssueRequestForUpdation(issue *UpdateIssueRequest, w http.ResponseWriter) bool {
```
This function should be in `utils`. Move all non-API functions to `utils`.

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	}
+
+	// submitting the request with an invalid milestone ID results in a 422 error
+	// we make sure it's not zero here, because the webapp client might have left this field empty
```
```suggestion
	// we should make sure it's not zero here because the webapp client might have left this field empty
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+
+	owner := splittedRepo[0]
+	repoName := splittedRepo[1]
+
```
```suggestion
	owner, repoName := splittedRepo[0], splittedRepo[1]
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	if post != nil {
+		_, appErr = p.API.CreatePost(reply)
+	} else {
+		p.API.SendEphemeralPost(c.UserID, reply)
```
```suggestion
		_ = p.API.SendEphemeralPost(c.UserID, reply)
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+
+	result, rawResponse, err := githubClient.Issues.CreateComment(c.Ctx, owner, repo, issueNumber, issueComment)
+	if err != nil {
+		statusCode := 500
```
```suggestion
		statusCode := http.StatusInternalServerError
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+		}
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to create an issue comment: %s", getFailReason(statusCode, repo, currentUsername)), StatusCode: statusCode})
+		return
+	}
```
```suggestion
	}
	
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	if post.RootId != "" {
+		// the original post was a reply
+		rootID = post.RootId
+	}
```
```suggestion
	}
	
```

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	if req.Status == Close {
+		p.CloseOrReopenIssue(c, w, "closed", req.StatusReason, req.Owner, req.Repository, post, issueNumber)
+	} else {
+		p.CloseOrReopenIssue(c, w, "open", req.StatusReason, req.Owner, req.Repository, post, issueNumber)
```
There should be constants for "closed" and "open"

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+    post: Post,
+}
+
+const GithubIssue = (props: GithubIssueProps) => {
```
```suggestion
const GithubIssue = ({theme, post}: GithubIssueProps) => {
```

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+            return style.close_or_reopen_button;
+        }
+        return null;
+    };
```
```suggestion
    };
    
```

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Made a package dor constants
2. Made a package for struct
3. Improved code redability

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+                >{postProps.status}</button>
+            </div>
+        </div>
+    );
```
```suggestion
    );
    
```

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+            </div>
+        </div>
+    );
+    if (postProps.assignees && postProps.assignees?.length !== 0) {
```
```suggestion
    if (postProps.assignees?.length) {
```

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+            <div style={style.assignees_and_labels}>
+                <b>{'Assignees'}</b>
+                <div>
+                    {postProps.assignees?.map((assignee: string, index: number) => (
```
```suggestion
                    {postProps.assignees.map((assignee: string, index: number) => (
```

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+                </div>
+            </div>
+        );
+    }
```
```suggestion
    }
    
```

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+            </div>
+        );
+    }
+    if (postProps.labels && postProps.labels?.length !== 0) {
```
Same as above.

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+};
+
+const getStyle = makeStyleFromTheme((theme) => {
+    return {
```
```suggestion
const getStyle = makeStyleFromTheme((theme) => ({
```

manojmalik20 commented on webapp/src/components/github_issue/github_issue.tsx:
```diff
@@ -0,0 +1,122 @@
+import {Theme} from 'mattermost-redux/types/preferences';
+import {Post} from 'mattermost-redux/types/posts';
+import {useDispatch} from 'react-redux';
+
```
Filename should be `index.tsx`.

manojmalik20 commented on webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx:
```diff
@@ -83,15 +107,16 @@ export default class AttachIssueModal extends PureComponent {
 
     render() {
-        const {visible, theme} = this.props;
         const {error, submitting} = this.state;
```
Extract all the variables that are needed below.

manojmalik20 commented on webapp/src/components/modals/close_reopen_issue/close_reopen_issue.tsx:
```diff
@@ -0,0 +1,182 @@
+
+import FormButton from '../../form_button';
+import Input from '../../input';
+
```
Change filename to `index.tsx`. Fix everywhere where applicable.

manojmalik20 commented on webapp/src/components/modals/close_reopen_issue/close_reopen_issue.tsx:
```diff
@@ -0,0 +1,182 @@
+    const dispatch = useDispatch();
+    const closeOrReopenIssueModalData = useSelector(getCloseOrReopenIssueModalData);
+    const messageData = closeOrReopenIssueModalData.messageData;
+    const visible = closeOrReopenIssueModalData.visible;
```
```suggestion
    const {messageData, visible} = closeOrReopenIssueModalData;
```

manojmalik20 commented on webapp/src/components/modals/close_reopen_issue/close_reopen_issue.tsx:
```diff
@@ -0,0 +1,182 @@
+                <label
+                    style={style.radioButtons}
+                    htmlFor='completed'
+                >{'Mark issue as completed'}
```
```suggestion
                >
                    {'Mark issue as completed'}
```

manojmalik20 commented on webapp/src/components/modals/close_reopen_issue/close_reopen_issue.tsx:
```diff
@@ -0,0 +1,182 @@
+                <label
+                    style={style.radioButtons}
+                    htmlFor='not_planned'
+                >{'Mark issue as not planned'}
```
Same.

manojmalik20 commented on webapp/src/components/modals/close_reopen_issue/close_reopen_issue.tsx:
```diff
@@ -0,0 +1,182 @@
+
+const getStyle = (theme: Theme) => ({
+    modal: {
+        padding: '2em 2em 3em',
```
Get all the CSS reviewed by a FE dev.

Nityanand13 replied:
Once it gets approved by you then I will ask any FE dev to review all the CSS.

manojmalik20 replied:
You shouldn't wait for my approval. Please get it reviewed by FE dev as soon as possible.

manojmalik20 commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,17 +46,35 @@ export default class CreateIssueModal extends PureComponent {
+            }});
+            this.setState({issueDescription: description});
+            this.setState({repo: repo_full_name});
+            this.setState({issueTitle: title.substring(0, MAX_TITLE_LENGTH)});
```
Call `setState` only once here.

manojmalik20 commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -80,20 +98,36 @@ export default class CreateIssueModal extends PureComponent {
-            return;
+        if (repo_full_name) {
+            const updated = await this.props.update(issue);
+            if (updated.error) {
```
```suggestion
            if (updated?.error) {
```

manojmalik20 commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -122,24 +156,28 @@ export default class CreateIssueModal extends PureComponent {
             return null;
         }
 
+        let repoName = this.state.repo.name;
+        if (!repoName) {
+            repoName = this.state.repo;
+        }
```
```suggestion
        const repoName = this.state.repo.name ?? this.state.repo;
```

manojmalik20 commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -176,7 +216,7 @@ export default class CreateIssueModal extends PureComponent {
         }
 
-        const component = (
+        let component = (
```
Instead of using `let`, use `const` and instead of `if` use conditional operators.

manojmalik20 commented on server/constants/constants.go:
```diff
@@ -0,0 +1,46 @@
+
+const (
+	APIErrorIDNotConnected = "not_connected"
+	// TokenTTL is the OAuth token expiry duration in seconds
+	TokenTTL = 10 * 60
```
How is this in seconds?
```suggestion
	// TokenTTL is the OAuth token expiry duration in minutes
	TokenTTL = 10 * time.Minute
```

Nityanand13 replied:
Is there any reason for this? As in `KVSetWithExpiry` we have to pass this in seconds.

manojmalik20 replied:
How many seconds is this? 600?

Nityanand13 replied:
Yes

manojmalik20 replied:
```suggestion
	// TokenTTL is the OAuth token expiry duration in seconds
	TokenTTL = 600 * time.Second
```

Nityanand13 replied:
I think we should not do this because `time.Second` is returning `1000000000`

manojmalik20 replied:
It's working fine in my system. Let's discuss this.

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Improved code quality
2. Fixed the issue that I was facing while creating issue or attaching a comment to the issue.
3. Changed few file names

manojmalik20 commented on server/serializer/issue.go:
```diff
@@ -36,9 +36,17 @@ type CommentAndCloseRequest struct {
 	PostID       string `json:"postId"`
 }
+
+type OpenCreateCommentOrEditIssueModal struct {
```
```suggestion
type OpenCreateCommentOrEditIssueModalRequestBody struct {
```

manojmalik20 commented on webapp/src/actions/index.js:
```diff
@@ -420,6 +420,15 @@ export function openCloseOrReopenIssueModal(messageData) {
     };
 }
 
+export function openCreateCommentIssueModal(messageData) {
```
```suggestion
export function openCreateCommentOnIssueModal(messageData) {
```

manojmalik20 commented on webapp/src/index.js:
```diff
@@ -4,18 +4,18 @@ import AttachCommentToIssuePostMenuAction from 'components/post_menu_actions/att
 
 import CreateOrUpdateIssueModal from './components/modals/create_update_issue';
-import CloseOrReopenIssueModal from './components/modals/close_reopen_issue/close_reopen_issue.tsx';
+import CloseOrReopenIssueModal from './components/modals/close_reopen_issue/index.tsx';
```
```suggestion
import CloseOrReopenIssueModal from './components/modals/close_reopen_issue;
```
When we don't specify a filename in the path, it takes the `index` file by default. Fix this wherever you've done this.

manojmalik20 commented on webapp/src/websocket/index.js:
```diff
@@ -105,11 +105,11 @@ export function handleOpenCloseOrReopenIssueModal(store) {
 }
 
-export function handleOpenAttachCommentIssueModal(store) {
+export function handleOpenCreateCommentIssueModal(store) {
```
```suggestion
export function handleOpenCreateCommentOnIssueModal(store) {
```

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Changed the names of few functions
2. Improved code quality

sooraj-shukla commented on webapp/src/client/client.js:
```diff
@@ -7,6 +7,18 @@ import {ClientError} from 'mattermost-redux/client/client4';
 import {id as pluginId} from '../manifest';
 
 export default class Client {
+    editIssueModal = async (payload) => {
+        return this.doPost(`${this.url}/editissuemodal`, payload);
+    }
+
+    closeOrReopenIssueModal = async (payload) => {
+        return this.doPost(`${this.url}/closereopenissuemodal`, payload);
+    }
+
+    attachCommentIssueModal = async (payload) => {
+        return this.doPost(`${this.url}/attachcommentissuemodal`, payload);
+    }
+
```
Can use single-line arrow function for all these.

Nityanand13 replied:
Will do in another PR.

sooraj-shukla commented on webapp/src/client/client.js:
```diff
@@ -7,6 +7,18 @@ import {ClientError} from 'mattermost-redux/client/client4';
 import {id as pluginId} from '../manifest';
 
 export default class Client {
+    editIssueModal = async (payload) => {
+        return this.doPost(`${this.url}/editissuemodal`, payload);
+    }
+
+    closeOrReopenIssueModal = async (payload) => {
+        return this.doPost(`${this.url}/closereopenissuemodal`, payload);
+    }
```
We aren't using await inside these, have we added `async` intentionally?

Nityanand13 replied:
Will do in another PR.

sooraj-shukla commented on webapp/src/client/client.js:
```diff
@@ -63,6 +75,14 @@ export default class Client {
         return this.doPost(`${this.url}/createissue`, payload);
     }
 
+    closeOrReopenIssue = async (payload) => {
+        return this.doPost(`${this.url}/closeorreopenissue`, payload);
+    }
+
+    updateIssue = async (payload) => {
+        return this.doPost(`${this.url}/updateissue`, payload);
+    }
+
```
Same as above two.

Nityanand13 replied:
Will do in another PR.

sooraj-shukla commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,129 @@
+
+    const content = (
+        <div>
+            <div>
```
Why are we wrapping the content inside two `divs`? We can use React fragments `<></>` or single `div` if it is required to wrap.

sooraj-shukla commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,129 @@
+                    className='btn btn-primary'
+                    onClick={() => {
+                        dispatch(attachCommentIssueModal(issue));
+                    }}
```
```suggestion
                    onClick={() => dispatch(attachCommentIssueModal(issue))}
```

sooraj-shukla commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,129 @@
+                    className='btn btn-primary'
+                    onClick={() => {
+                        dispatch(editIssueModal(issue));
+                    }}
```
```suggestion
                    onClick={() => dispatch(editIssueModal(issue))}
```

sooraj-shukla commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,129 @@
+                    className={buttonClassName}
+                    onClick={() => {
+                        dispatch(closeOrReopenIssueModal(issue));
+                    }}
```
Same as above.

sooraj-shukla commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -0,0 +1,129 @@
+                    }}
+                >{'Comment'}</button>
+                <button
+                    style={{...style.button, ...style.other_buttons}}
```
Can't we use class names instead of inline styles?

sooraj-shukla commented on webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx:
```diff
@@ -83,29 +107,40 @@ export default class AttachIssueModal extends PureComponent {
+                    onChange={this.handleIssueCommentChange}
+                    value={comment}
+                />
+
```
```suggestion
```

sooraj-shukla commented on webapp/src/components/modals/close_reopen_issue/index.tsx:
```diff
@@ -0,0 +1,181 @@
+
+const getStyle = (theme: Theme) => ({
+    modal: {
+        padding: '2em 2em 3em',
```
Use consistent spacing unit(px).

sooraj-shukla commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -176,11 +212,40 @@ export default class CreateIssueModal extends PureComponent {
+                />
+
+                <Input
+                    id={'title'}
```
```suggestion
                    id='title'
```

sooraj-shukla commented on webapp/src/components/sidebar_buttons/sidebar_buttons.jsx:
```diff
@@ -208,28 +208,26 @@ export default class SidebarButtons extends React.PureComponent {
+    },
+    buttonHeader: {
+        color: changeOpacity(theme.sidebarText, 0.6),
+        textAlign: 'center',
+        cursor: 'pointer',
```
See if we have util classes for these.

Nityanand13 replied:
Will do in another PR.

sooraj-shukla commented on webapp/src/components/sidebar_buttons/sidebar_buttons.jsx:
```diff
@@ -208,28 +208,26 @@ export default class SidebarButtons extends React.PureComponent {
+    containerHeader: {
+        marginTop: '10px',
+        marginBottom: '5px',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-around',
```
See if we've util classes for these.

Nityanand13 replied:
Will do in another PR.

sooraj-shukla commented on webapp/src/components/sidebar_buttons/sidebar_buttons.jsx:
```diff
@@ -208,28 +208,26 @@ export default class SidebarButtons extends React.PureComponent {
+        justifyContent: 'space-around',
+        padding: '0 10px',
+    },
+    containerTeam: {
+    },
```
Please remove if not required.

Nityanand13 replied:
Will do in another PR.

sooraj-shukla commented on webapp/src/components/sidebar_right/github_items.tsx:
```diff
@@ -255,58 +255,56 @@ function GithubItems(props: GithubItemsProps) {
+        fontWeight: 'bold',
+    },
+    subtitle: {
+        margin: '5px 0 0 0',
```
marginTop

Nityanand13 committed:
[MI-2502]: Review fixes done
1.Improved code readability

sooraj-shukla commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -21,11 +21,9 @@ const GithubIssue = ({theme, post}: GithubIssueProps) => {
         }
-        return null;
+        return style.other_buttons;
     };
```
We aren't doing anything here, simply returning the variables. We can get rid of this function if not needed.

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Improved code readability

manojmalik20 commented on webapp/src/components/github_issue/index.tsx:
```diff
@@ -114,7 +104,7 @@ const getStyle = makeStyleFromTheme((theme) => ({
     },
     close_or_reopen_button: {
-        backgroundColor: theme.buttonBg,
+        backgroundColor: '#dc3545',
```
Why did you change this?

Nityanand13 replied:
Basically this is the default colour of danger.So, if we have `Close` status then we are showing this colour otherwise `theme.buttonBg`

https://github.com/Brightscout/mattermost-plugin-github/blob/0a8db2e5b68badf7d0d26400a18ce9412f398f36/webapp/src/components/github_issue/index.tsx#L43

manojmalik20 replied:
There must be variable for color `danger` also. Use that.

Nityanand13 replied:
I got an `errorTextColor` similar to this colour so used that.

Nityanand13 committed:
[MI-2502]: Review done
1. Improved code quality

Nityanand13 committed:
[MI-2502]: Review fixes done
1. Improved code quality
